src/app/account-app/payments.service.ts

```diff
@@ -0,0 +1,65 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2019 Runbox Solutions AS (runbox.com).
+//
+// This file is part of Runbox 7.
+//
+// Runbox 7 is free software: You can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+import { RunboxWebmailAPI } from '../rmmapi/rbwebmail';
+
+import { Injectable } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of, Observable, Subject, ReplaySubject } from 'rxjs';
+import { Product } from './product';
+import { Cart } from './cart';
+
+@Injectable()
+export class PaymentsService {
+    errorLog = new Subject<HttpErrorResponse>();
+
+    products       = new ReplaySubject<Product[]>(1);
+    activeProducts = new ReplaySubject<any>(1);
+    stripePubkey   = new ReplaySubject<string>(1);
+    currency       = new ReplaySubject<string>(1);
+
+    cart: Cart = new Cart();
+
+    constructor(
+        private rmmapi: RunboxWebmailAPI,
+    ) {
+        this.rmmapi.getAvailableProducts().subscribe(products => {
+            this.products.next(products);
```

Comment:
ReplaySubject or AsyncSubject? If there's only one event to be pushed here in the app lifetime, I would probably rather use a completing AsyncSubject.

Reply:
Good question; I think 3/4 of those are “immutable”, while `activeProducts` may change (whenever you buy something). That said, we never refresh it anywhere, so there is currently no point in them being `ReplaySubject`s. So if I use `AsyncSubject` here I should also do the `.complete()` later – but will future subscribers still get the value that we obtained here?

Reply:
Addressed in fd65e0b.